activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -109,19 +109,21 @@ def dirty?
         @dirty
       end
 
-      def add_record(record, ensure_finalize = true)
-        @records ||= []
+      # Add the record to the current transaction, used by after_commit (etc) callbacks.
+      # Returns true if the record was added to the transaction, false if it had been previously added.
+      def uniquely_add_record(record, ensure_finalize = true)
+        @records ||= Set.new.compare_by_identity
         if ensure_finalize
-          @records << record
+          @records.add(record)
```

Comment:
This needs to be `add?` to get the boolean return... which means we're missing a test.

Reply:
I fixed that up but I'm struggling to see where the missing test should go :/